### PR TITLE
feat: Port to msfs-geo plus few other fixes

### PR DESCRIPTION
**Pull request by professoralex13**
- Fixes Faulty ils location longitude
- Disables no-bitwise eslint rule

### CODE REVIEW

File: src/server/providers/navigraph_dfd/mappers.ts [hunk 1]
```diff
@@ -1139,16 +1136,20 @@ export class DFDMappers {
...
+        return { lat, long };
+    }
+
+    public static mapElevatedCoordinates(lat: number, long: number, alt: number): ElevatedCoordinates {
+        return { lat, long, alt };
     }
 
     private static isTerminalWaypoint(waypoint: NaviWaypoint): waypoint is NaviTerminalWaypoint {
-        return waypoint.hasOwnProperty('regionCode') && waypoint['regionCode'] !== null;
+        return waypoint.hasOwnProperty('regionCode') && waypoint.regionCode !== null;
```
Comment (tracernz) [inline src/server/providers/navigraph_dfd/mappers.ts]: The linter did not like this... you sure it does now?

File: src/server/providers/navigraph_dfd/dfd.ts [hunk 2]
```diff
@@ -428,29 +424,28 @@ export class NavigraphProvider implements DataInterface {
...
+        const longitudeCrosses = southWestCorner.long > northEastCorner.long;
 
         let sql: string;
         let params: any[];
 
         if (longitudeCrosses) {
             sql = `${prefix}latitude >= ? AND ${prefix}latitude <= ? AND (${prefix}longitude >= ? OR ${prefix}longitude <= ?)`;
-            params = [southWestCorner.latitude, northEastCorner.latitude, southWestCorner.longitude, northEastCorner.longitude];
-        } else if (Math.max(southWestCorner.latitude, northEastCorner.latitude) > 80) {
+            params = [southWestCorner.lat, northEastCorner.lat, southWestCorner.lat, northEastCorner.lat];
```
Comment (tracernz) [inline src/server/providers/navigraph_dfd/dfd.ts]:
```suggestion
            params = [southWestCorner.lat, northEastCorner.lat, southWestCorner.long, northEastCorner.long];
```

File: src/shared/types/GlsNavaid.ts [hunk 3]
```diff
@@ -1,10 +1,11 @@
-import { DatabaseItem, Degrees, DegreesMagnetic, Location, LsCategory } from './Common';
+import { Coordinates, Degrees, DegreesMagnetic } from 'msfs-geo';
+import { DatabaseItem, LsCategory } from './Common';
 
 export interface GlsNavaid extends DatabaseItem {
     channel: number;
     category: LsCategory;
     runwayIdent: string;
-    location: Location;
+    location: Coordinates;
```
Comment (tracernz) [inline src/shared/types/GlsNavaid.ts]: This should have an elevation, else we can't simulate the glideslope.

File: src/shared/types/IlsNavaid.ts [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 
 export interface IlsNavaid extends DatabaseItem {
     frequency: KiloHertz;
     category: LsCategory;
     runwayIdent: string;
-    locLocation: Location;
+    locLocation: Coordinates;
     locBearing: DegreesMagnetic;
-    gsLocation?: Location;
+    gsLocation?: Coordinates;
```
Comment (tracernz) [inline src/shared/types/IlsNavaid.ts]: Needs to be ElevatedCoordinates | Coordinates.

File: src/shared/types/VhfNavaid.ts [hunk 5]
```diff
@@ -11,8 +12,8 @@ export interface VhfNavaid extends DatabaseItem {
      * Beware: this is NOT the same as magnetic variation
      */
     stationDeclination: Degrees,
-    vorLocation: Location,
-    dmeLocation?: Location,
+    vorLocation: Coordinates,
+    dmeLocation?: ElevatedCoordinates,
```
Comment (tracernz) [inline src/shared/types/VhfNavaid.ts]: Needs to be ElevatedCoordinates | Coordinates

**Review by tracernz — CHANGES_REQUESTED** [review]

File: src/shared/types/IlsNavaid.ts (hunk 4 above)
Comment (professoralex13) [reply src/shared/types/IlsNavaid.ts]: we can just set the elevation to the airport elevation if the ils elevation isnt defined right?

File: src/server/providers/navigraph_dfd/mappers.ts [hunk 6]
```diff
@@ -1145,11 +1146,15 @@ export class DFDMappers {
     }
 
     private static isTerminalWaypoint(waypoint: NaviWaypoint): waypoint is NaviTerminalWaypoint {
-        return waypoint.hasOwnProperty('regionCode') && waypoint.regionCode !== null;
+        // @ts-ignore
+        // eslint-disable-next-line no-prototype-builtins,dot-notation
```
Comment (tracernz) [inline src/server/providers/navigraph_dfd/mappers.ts:1149]: Why do this when it was happy before you did anything? 🤣

File: .eslintrc.js [hunk 7]
```diff
@@ -10,5 +10,6 @@ module.exports = {
     rules: {
         "no-undef": "off",
         "import/no-unresolved": "off",
+        "no-bitwise": "off",
```
Comment (tracernz) [inline .eslintrc.js:13]: We should do this upstream eventually. Silly rule.

**Review by tracernz — APPROVED** [review]

**Pull request merged**